packages/nimble-components/src/table/components/header/tests/table-header.spec.ts

```diff
@@ -94,4 +94,18 @@ describe('TableHeader', () => {
         expect(sortIcons.ascendingIcon).toBeFalsy();
         expect(sortIcons.descendingIcon).toBeTruthy();
     });
+
+    it('displays grouping indicator icon when grouped', async () => {
+        element.isGrouped = true;
+        await waitForUpdatesAsync();
+
+        const groupingIndicatorIcon = element.shadowRoot?.querySelector('.grouped-indicator');
```

Comment:
We don't have strict guidance about what belongs in page objects in this repo since it's a fairly new pattern, but my default suggestion is to put anything that does a `querySelector` in a page object so that tests don't need to change if the DOM structure changes. And based on previous guidance, I'd have that page object return a boolean for whether the icon is found rather than returning the element itself.